internal/awsdata/ecs.go

```diff
@@ -0,0 +1,163 @@
+		var taskArns []*string
+		done := false
+		params := &ecs.ListTasksInput{
+			Cluster: cluster.ClusterArn,
+		}
+
+		for !done {
+			outListTasks, err := ecsSvc.ListTasks(params)
+			if err != nil {
+				d.results <- result{Err: err}
+				return
+			}
+
+			taskArns = append(taskArns, outListTasks.TaskArns...)
+
+			if outListTasks.NextToken == nil {
+				done = true
+			} else {
+				params.NextToken = outListTasks.NextToken
+			}
+		}
+
+		// TODO: API call can only handle 100 task ARNs at a time
+		outDescribeTasks, err := ecsSvc.DescribeTasks(&ecs.DescribeTasksInput{
+			Cluster: cluster.ClusterArn,
+			Tasks: taskArns,
+		})
+		if err != nil {
+			d.results <- result{Err: err}
+			return
+		}
+		for _, task := range outDescribeTasks.Tasks {
+			for _, container := range task.Containers {
+				d.wg.Add(1)
+				go d.processECSContainer(container, task, cluster, region)
+			}
+		}
+	}
+
+	log.Info("finished processing instance data")
+}
+
+func (d *AWSData) processECSContainer(container *ecs.Container, task *ecs.Task, cluster *ecs.Cluster, region string) {
+	defer d.wg.Done()
+
+	var ips []string
+	var macAddresses []string
+	var networkInterfaces []string
+
+	for _, attachment := range task.Attachments {
+		if aws.StringValue(attachment.Type) != "ElasticNetworkInterface" {
+			continue
+		}
+		for _, details := range attachment.Details {
+			switch aws.StringValue(details.Name) {
+			case "privateIPv4Address":
+			  ips = append(ips, aws.StringValue(details.Value))
+			case "ipv6Address":
+			  ips = append(ips, aws.StringValue(details.Value))
+			case "macAddress":
+			  macAddresses = append(macAddresses, aws.StringValue(details.Value))
+			case "networkInterfaceId":
+			  networkInterfaces = append(networkInterfaces, aws.StringValue(details.Value))
+			}
+		}
+	}
+
+	for _, networkInterface := range container.NetworkInterfaces {
+		ips = appendIfMissing(ips, aws.StringValue(networkInterface.PrivateIpv4Address))
+		ips = appendIfMissing(ips, aws.StringValue(networkInterface.Ipv6Address))
+	}
+
+	var hardware = aws.StringValue(task.LaunchType)
+	if hardware == "FARGATE" {
+		hardware = fmt.Sprintf("%s %s", hardware, aws.StringValue(task.PlatformVersion))
+	}
+
+	d.results <- result{
+		Row: inventory.Row{
+			UniqueAssetIdentifier:     fmt.Sprintf("%s-%s", aws.StringValue(container.Name), aws.StringValue(container.RuntimeId)),
+			IPv4orIPv6Address:         strings.Join(ips, "\n"),
+			Virtual:                   true,
+			MACAddress:                strings.Join(macAddresses, "\n"),
+			BaselineConfigurationName: aws.StringValue(container.Image),
+			Location:                  region,
+			AssetType:                 AssetTypeECSContainer,
+			HardwareMakeModel:         hardware,
+			Function:                  fmt.Sprintf("%s %s", aws.StringValue(cluster.ClusterName), aws.StringValue(task.Group)),
+			SerialAssetTagNumber:      aws.StringValue(container.ContainerArn),
+			// TODO: Query EC2 API to map ENI to VPC
```

Comment:
Let me know if you know an easier way to get this. The UI in the console shows it, but I can't find an easy way to get it via the API (unless perhaps I use the EC2 API).

Reply:
Hmm I can't see another way at the moment. I think that's because of the ECS design where the actual infrastructure the container run on is separate. I guess the only way would be the EC2 API

Reply:
I went ahead and added support for pulling this from the EC2 API.